java/code/src/com/redhat/rhn/frontend/xmlrpc/serializer/UserNotificationSerializer.java

```diff
@@ -0,0 +1,47 @@
+package com.redhat.rhn.frontend.xmlrpc.serializer;
+
+import com.redhat.rhn.domain.org.usergroup.UserExtGroup;
+import com.redhat.rhn.domain.role.Role;
+import com.redhat.rhn.domain.notification.UserNotification;
+
+import com.suse.manager.api.ApiResponseSerializer;
+import com.suse.manager.api.SerializationBuilder;
+import com.suse.manager.api.SerializedApiResponse;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *ff
+ * UserNotificationSerializer
+ *
+ * @apidoc.doc
+ *  #struct_begin("notification")
+ *      #prop("long", "id")
+ *      #prop("long", "messageId")
+ *      #prop_array("boolean", "read")
+ *      #prop_array("string", "message")
+ *      #prop_array("notificationType", "type")
+ *      #prop_array("date", "created")
+ *  #struct_end()
+ *
+ */
+public class UserNotificationSerializer extends ApiResponseSerializer<UserNotification> {
+
+    @Override
+    public Class<UserNotification> getSupportedClass() {
+        return UserNotification.class;
+    }
+
+    @Override
+    public SerializedApiResponse serialize(UserNotification src) {
+        return new SerializationBuilder()
+                .add("id", src.getId())
+                .add("messageId", src.getMessage().getId())
```

Comment:
`messageId` does not belong to the "User" domain and it has no use for the user, so I think we can leave it out (also from the javadoc above).